packages/synapse-react-client/src/synapse-queries/table/useDefiningSql.ts

```diff
@@ -0,0 +1,18 @@
+import { SynapseClientError } from '../../utils'
+import { UseMutationOptions, useMutation } from '@tanstack/react-query'
+import { validateDefiningSql } from '../../synapse-client'
+import {
+  ValidateDefiningSqlResponse,
+  EntityType,
+} from '@sage-bionetworks/synapse-types'
+
+export function useValidateDefiningSql(
+  sql: string,
+  entityType: EntityType,
+  options?: UseMutationOptions<ValidateDefiningSqlResponse, SynapseClientError>,
+) {
+  return useMutation<ValidateDefiningSqlResponse, SynapseClientError>({
+    ...options,
+    mutationFn: () => validateDefiningSql(sql, entityType),
+  })
+}
```

Comment:
This doesn't look right, the `sql` and `entityType` args should be passed to the `mutationFn`, not the hook